tests/common_templates_test.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"flag"
+	"os/exec"
+	"strings"
+	"net/http"
+	"io/ioutil"
+	"regexp"
+	"os"
+	"bufio"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Common templates", func() {
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	// Common templates read into string
+	var common_templates_yaml string
+
+	BeforeEach(func() {
+		tests.BeforeTestCleanup()
+
+		// Setting namespace
+		_, err := exec.Command("/bin/bash", "-c", "/usr/bin/oc project " + tests.NamespaceTestDefault).Output()
+		tests.PanicOnError(err)
+
+		// Applying datavolume
+		// TODO1: This is currently bugged and replaced with using vm.yaml. 
+		// exec.Command("/bin/bash", "-c", "/usr/bin/oc apply -f tests/manifests/fedora_datavolume.yaml")
+
+		// Getting common_templates
+		// TODO2: replace downloading common-templates with getting common-templates from RPM
+		ct_yml_url := "https://github.com/kubevirt/common-templates/releases/download/v0.3.1/common-templates-v0.3.1.yaml"
```

Comment:
What happens when we release a new version of templates? Who will update the test case?

Reply:
The last version should be tested.
I've added two functions to utils that obtain the latest version of common templates and now this test uses these functions instead of using hardcoded value. Here is the merge request for utils:
https://github.com/kubevirt/kubevirt-ansible/pull/531